From 4fbed43077bb248cda7af30440a21198d1f87815 Mon Sep 17 00:00:00 2001
From: Michael Haubenwallner <michael.haubenwallner@ssi-schaefer.com>
Date: Tue, 26 Nov 2019 15:05:45 +0100
Subject: [PATCH 6/8] libtool: find NAME.lib with -lNAME eventually

With MSVC like compilers, libname_spec may omit the lib prefix, and we
may use $libext even for the dynamic import library.  When the package
manager does build the shared library only and subsequently does remove
the .la file, we won't find any library with -lNAME.

* build-aux/ltmain.in (func_mode_link): Use $library_names_spec in
addition to the traditional file names when searching for library files
upon the -lNAME linker option.
---
 build-aux/ltmain.in | 30 ++++++++++++++++++++++++------
 1 file changed, 24 insertions(+), 6 deletions(-)

diff --git a/build-aux/ltmain.in b/build-aux/ltmain.in
index 73c51d06..19257a55 100644
--- a/build-aux/ltmain.in
+++ b/build-aux/ltmain.in
@@ -5787,17 +5787,26 @@ func_mode_link ()
 	  fi
 	  func_stripname '-l' '' "$deplib"
 	  name=$func_stripname_result
+	  release=
+	  major=
+	  versuffix=
+	  module_save=$module
+	  module=no
+	  eval shared_ext=\"$shrext_cmds\"
+	  eval libname=\"$libname_spec\"
+	  eval search_names=\"lib$name.la $library_names_spec lib$name$shared_ext lib$name.so $libname.$libext lib$name.$libext lib$name.a\"
+	  module=$module_save
 	  if test lib = "$linkmode"; then
 	    searchdirs="$newlib_search_path $lib_search_path $compiler_lib_search_dirs $sys_lib_search_path $shlib_search_path"
 	  else
 	    searchdirs="$newlib_search_path $lib_search_path $sys_lib_search_path $shlib_search_path"
 	  fi
 	  for searchdir in $searchdirs; do
-	    for search_ext in .la $std_shrext .so .a; do
+	    for search_name in $search_names; do
 	      # Search the libtool library
-	      lib=$searchdir/lib$name$search_ext
+	      lib=$searchdir/$search_name
 	      if test -f "$lib"; then
-		if test .la = "$search_ext"; then
+		if test "lib$name.la" = "$search_name"; then
 		  found=:
 		else
 		  found=false
@@ -7486,7 +7495,12 @@ EOF
 		esac
 	      fi
 	      if test -n "$a_deplib"; then
-		libname=`eval "\\$ECHO \"$libname_spec\""`
+		module_save=$module
+		module=no
+		eval shared_ext=\"$shrext_cmds\"
+		eval libname=\"$libname_spec\"
+		eval search_names=\"$library_names_spec\"
+		module=$module_save
 		if test -n "$file_magic_glob"; then
 		  libnameglob=`func_echo_all "$libname" | $SED -e $file_magic_glob`
 		else
@@ -7494,12 +7508,16 @@ EOF
 		fi
 		test yes = "$want_nocaseglob" && nocaseglob=`shopt -p nocaseglob`
 		for i in $lib_search_path $sys_lib_search_path $shlib_search_path; do
+		  potential_libs=
+		  for search_name in $search_names; do
+		    potential_libs="$potential_libs `ls $i/$search_name 2>/dev/null`"
+		  done
 		  if test yes = "$want_nocaseglob"; then
 		    shopt -s nocaseglob
-		    potential_libs=`ls $i/$libnameglob[.-]* 2>/dev/null`
+		    potential_libs="$potential_libs `ls $i/$libnameglob[.-]* 2>/dev/null`"
 		    $nocaseglob
 		  else
-		    potential_libs=`ls $i/$libnameglob[.-]* 2>/dev/null`
+		    potential_libs="$potential_libs `ls $i/$libnameglob[.-]* 2>/dev/null`"
 		  fi
 		  for potent_lib in $potential_libs; do
 		      # Follow soft links.
-- 
2.22.0

